### ISSUE TITLE
mesh can not have more than 65000 vertices

### ISSUE THREAD

**Issue by Bod9001** [unitystation/unitystation#3055]
looks like there's a vertices leak somewhere, the text brokef for them, I don't know how to reproduce

**Comment by Bod9001** [comment]
is Chat spam, Invalid characters may be

**Comment by Misterixmax** [comment]
Mesh vertices? What the hell. It looks more like someone spawned a highly detailed object, really detailed.

**Comment by Macoron** [comment]
Nope, it's because chat actually generated as mesh. It's a  Unity way to represent UI.

**Comment by Alainx277** [comment]
Take a look at Mesh.indexFormat. If you want more than 65000 vertices it needs to be IndexFormat.UInt32

**Comment by ghost** [comment]
More information on this issue here:
https://forum.unity.com/threads/ui-text-character-limit.359729/

Would it be possible to cull text from the canvas beyond a certain limit and redraw on scrollup events?

**Comment by Alainx277** [comment]
> More information on this issue here:
> https://forum.unity.com/threads/ui-text-character-limit.359729/

Seems like it would be a good idea to use Text Mesh Pro

**Comment by jamessimo** [comment]
Don’t we use text mesh pro now? TMP breaks up objects  if they bypass this 65k verts

**Comment by david-pfx** [comment]
TMP comes with a heap of issues if all you need is plain text beyond a few hundred lines. Actually, it's a horrible fit for this use case.

Would it be so hard to rebuild Text in UInt32?
